keys/keystore.go

```diff
@@ -66,12 +65,21 @@ func decryptKey(keyProtected *EncryptedKeyJSON, auth string) ([]byte, error) {
 	bufferValue := make([]byte, len(cipherText)+16)
 	copy(bufferValue[0:16], derivedKey[16:32])
 	copy(bufferValue[16:], cipherText[:])
-	calculatedMAC := sha256.Sum256([]byte((bufferValue)))
+	hasher := sha3.NewLegacyKeccak512()
```

Comment:
please change to calculate hash according to `kdfparams.prf` in keystore.json in future